packages/components/src/components/rating/rating.jsx

```diff
@@ -0,0 +1,134 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import Text from '../text';
+
+const Rating = ({
+    allow_half_rating = false,
+    background_class,
+    direction = 'ltr',
+    disable_hover = false,
+    default_content,
+    element_class,
+    icon_selected,
+    icon_unselected,
+    max_rating = 5,
+    onClick,
+    readonly = false,
+    show_rating = false,
+    value,
+}) => {
+    const [enabled_rate_index, setEnabledRateIndex] = React.useState(-1);
+    const [hovered_rate_index, setHoveredRateIndex] = React.useState(-1);
+    const [is_hover_enabled, setIsHoverEnabled] = React.useState(false);
+    const ratingContainerRef = React.useRef(null);
+
+    const calculateRating = e => {
+        const { width, left, right } = ratingContainerRef.current.getBoundingClientRect();
+        const rating_start_element = direction === 'ltr' ? e.clientX - left : right - e.clientX;
+        const selected_rating = (rating_start_element / width) * max_rating;
+        if (selected_rating > 0.4) {
+            const calculated_rating = allow_half_rating
+                ? Math.round((selected_rating + 0.5 / 2) / 0.5) * 0.5 // This will always make sure that, if we hover between two ratings, it will select the nearest number
+                : Math.round(selected_rating);
+            return calculated_rating;
+        }
+        return 0;
```

Comment:
why do we need to check if `selected_rating` is above 0.4?

```suggestion
            const calculated_rating = allow_half_rating
                ? Math.round(selected_rating / 0.5) * 0.5
                : Math.round(selected_rating);
            return calculated_rating;
```

Reply:
// Some calculation examples

Math.round(1.43 / 0.5) * 0.5  = 1.5
Math.round(0.26 / 0.5) * 0.5  = 0.5
Math.round(0.2 / 0.5) * 0.5  = 0

Reply:
Thank you. I will make the change